server/src/main/java/org/opensearch/index/store/RemoteDirectory.java

```diff
@@ -152,7 +155,14 @@ public IndexInput openInput(String name, IOContext context) throws IOException {
             return new RemoteIndexInput(name, inputStream, fileLength(name));
         } catch (Exception e) {
             // Incase the RemoteIndexInput creation fails, close the input stream to avoid file handler leak.
-            if (inputStream != null) inputStream.close();
+            if (inputStream != null) {
+                try {
+                    inputStream.close();
+                } catch (Exception closeEx) {
+                    e.addSuppressed(closeEx);
+                }
+            }
+            logger.error("Exception while reading blob for file: " + name + " for path " + blobContainer.path());
             throw e;
```

Comment:
Lets log `e`

Reply:
Should this be used as try with resource?

Reply:
This is getting logged in the upstream.